compile/inc/src/main/scala/sbt/inc/APIs.scala

```diff
@@ -38,6 +38,8 @@ trait APIs {
 
   def internal: Map[File, Source]
   def external: Map[String, Source]
+
+  def hasPackageObject(src: File): Boolean
```

Comment:
I think it only breaks forward compatibility not backwards compatibility?

Also even if it's a breaking change that doesn't necessarily make it a blocker. sbt exposes too much and we've made concessions before. It's partially why we don't use MiMa currently in 0.13.